static/public_pages/utils/photometry.js

```diff
@@ -1,204 +1,198 @@
-/* global Plotly */
-const baseLayout = {
-  zeroline: false,
-  automargin: true,
-  showline: true,
-  autorange: "reversed",
-  titlefont: { size: 18 },
-  tickfont: { size: 14 },
-  ticklen: 12,
-  ticks: "outside",
-  nticks: 8,
-  minor: {
-    ticks: "outside",
-    ticklen: 6,
-    tickcolor: "black",
-  },
-};
-
-/* eslint-disable */
-function getHoverText(point) {
-  return (
-    `MJD: ${point.mjd.toFixed(6)}<br>` +
-    (point.mag !== null ? `Mag: ${point.mag.toFixed(4)}<br>` : "") +
-    (point.magerr !== null ? `Magerr: ${point.magerr.toFixed(4)}<br>` : "") +
-    (point.limiting_mag !== null
-      ? `Limiting Mag: ${point.limiting_mag.toFixed(4)}<br>`
-      : "") +
-    `Filter: ${point.filter}<br>Instrument: ${point.instrument_name}`
-  );
-}
-
-function getTrace(data, isDetection, key, color, isMobile) {
-  const now = new Date().getTime() / 86400000 + 40587;
-  const rgba = (rgb, alpha) => `rgba(${rgb[0]},${rgb[1]},${rgb[2]}, ${alpha})`;
-  const dataType = isDetection ? "detections" : "upperLimits";
-  return {
-    dataType,
-    x: data.map((point) => now - point.mjd),
-    y: data.map((point) => (isDetection ? point.mag : point.limiting_mag)),
-    ...(isDetection
-      ? {
-          error_y: {
-            type: "data",
-            array: data.map((point) => point.magerr),
-            visible: true,
-            color: rgba(color, 0.5),
-            width: 1,
-            thickness: 2,
-          },
-        }
-      : {}),
-    text: data.map((point) => getHoverText(point)),
-    mode: "markers",
-    type: "scatter",
-    name: key + (isDetection ? "" : " (UL)"),
-    legendgroup: key + dataType,
-    marker: {
-      line: {
-        width: 1,
-        color: rgba(color, 1),
+function photometryPlot(
+  photometry_data,
+  div_id,
+  filters_used_mapper,
+  isMobile,
+) {
+  function getBaseLayout() {
+    return {
+      zeroline: false,
```

Comment:
Maybe I am missing something (these things can be hard to read on git's diff UI), but why do we need to nest these functions exactly?